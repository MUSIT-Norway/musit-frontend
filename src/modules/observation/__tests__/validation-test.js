import {
  validateFromTo,
  validateAlcohol,
  validatePest,
  validateDoubleTextArea
} from '../observationValidation';

describe('Validation.validateAlcohol', () => {
  it('should verify that statusValue cannot be null', () => {
    const formProps = {
      statusValue: null
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that statusValue cannot be empty', () => {
    const formProps = {
      statusValue: ''
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that statusValue cannot be empty space', () => {
    const formProps = {
      statusValue: ' '
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that volumeValue cannot be greater than 10', () => {
    const formProps = {
      statusValue: 'something',
      volumeValue: '11111111111,34'
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that volumeValue cannot contain dots', () => {
    const formProps = {
      statusValue: 'something',
      volumeValue: '11.34'
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should validate that commentsValue cannot be longer than 250 chars', () => {
    const formProps = {
      statusValue: 'something',
      volumeValue: '11,34',
      commentValue: 'x'.repeat(251)
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should validate correct values', () => {
    const formProps = {
      statusValue: 'something',
      volumeValue: '11,34',
      commentValue: 'x'.repeat(250)
    };
    const errors = validateAlcohol(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });
});

describe('Validation.validateFromTo', () => {
  it('should verify that fromValue cannot be null', () => {
    const formProps = {
      fromValue: null
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that fromValue cannot be empty', () => {
    const formProps = {
      fromValue: ''
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that fromValue cannot be empty space', () => {
    const formProps = {
      fromValue: ' '
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that toValue must be greater than fromValue', () => {
    const formProps = {
      fromValue: '42,65',
      toValue: '42,40'
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should not accept fromValue or toValue with dots as comma', () => {
    const formProps = {
      fromValue: '44.65',
      toValue: '48.40'
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should validate that commentsValue cannot be longer than 250 chars', () => {
    const formProps = {
      fromValue: '44,65',
      toValue: '48,40',
      commentValue: 'x'.repeat(251)
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should validate correct values', () => {
    const formProps = {
      fromValue: '44,65',
      toValue: '48,40',
      commentValue: 'x'.repeat(250)
    };
    const errors = validateFromTo(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });
});

describe('Validation.validatePest', () => {
  it('should verify that identificationValue cannot be null', () => {
    const formProps = {
      identificationValue: null,
      observations: []
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that identificationValue cannot be empty', () => {
    const formProps = {
      identificationValue: '',
      observations: []
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that identificationValue cannot be empty space', () => {
    const formProps = {
      identificationValue: ' ',
      observations: []
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that commentValue cannot be longer than 250 chars', () => {
    const formProps = {
      identificationValue: 'something',
      commentValue: 'x'.repeat(251),
      observations: []
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify correct values', () => {
    const formProps = {
      identificationValue: 'something',
      commentValue: 'x'.repeat(250),
      observations: [
        {
          count: 11,
          lifeCycle: 'flies'
        }
      ]
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that an observation with count and not lifecycle is not allowed', () => {
    const formProps = {
      identificationValue: 'something',
      commentValue: 'x'.repeat(250),
      observations: [
        {
          count: 1
        }
      ]
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that an observation with count larger than 10 numbers is not allowed', () => {
    const formProps = {
      identificationValue: 'something',
      commentValue: 'x'.repeat(250),
      observations: [
        {
          lifeCycle: 'something',
          count: '11111111111'
        }
      ]
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that an observation with only lifecycle is allowed', () => {
    const formProps = {
      identificationValue: 'something',
      commentValue: 'x'.repeat(250),
      observations: [
        {
          lifeCycle: 'something'
        }
      ]
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should verify that an observation with empty lifeCycle is not allowed', () => {
    const formProps = {
      identificationValue: 'something',
      commentValue: 'x'.repeat(250),
      observations: [
        {
          lifeCycle: ' '
        }
      ]
    };
    const errors = validatePest(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });
});

describe('Validate.validateDoubleTextArea', () => {
  it('should work with correct data', () => {
    const formProps = {
      leftValue: 'something',
      rightValue: 'something else'
    };
    const errors = validateDoubleTextArea(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should fail with no leftValue', () => {
    const formProps = {
      leftValue: null
    };
    const errors = validateDoubleTextArea(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });

  it('should fail with empty space leftValue', () => {
    const formProps = {
      leftValue: ' '
    };
    const errors = validateDoubleTextArea(formProps, 0, 'type');
    expect(errors).toMatchSnapshot();
  });
});
