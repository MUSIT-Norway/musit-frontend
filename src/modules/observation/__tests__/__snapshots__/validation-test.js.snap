// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate.validateDoubleTextArea should fail with empty space leftValue 1`] = `
Object {
  "type.leftValue": "musit.observation.page.type.leftValueIncorrect",
}
`;

exports[`Validate.validateDoubleTextArea should fail with no leftValue 1`] = `
Object {
  "type.leftValue": "musit.observation.page.type.leftValueRequired",
}
`;

exports[`Validate.validateDoubleTextArea should work with correct data 1`] = `Object {}`;

exports[`Validation.validateAlcohol should validate correct values 1`] = `Object {}`;

exports[`Validation.validateAlcohol should validate that commentsValue cannot be longer than 250 chars 1`] = `
Object {
  "type.commentValue": "musit.observation.page.type.commentValueIncorrect",
}
`;

exports[`Validation.validateAlcohol should verify that statusValue cannot be empty 1`] = `
Object {
  "type.statusValue": "musit.observation.page.type.statusValueRequired",
}
`;

exports[`Validation.validateAlcohol should verify that statusValue cannot be empty space 1`] = `
Object {
  "type.statusValue": "musit.observation.page.type.statusValueIncorrect",
}
`;

exports[`Validation.validateAlcohol should verify that statusValue cannot be null 1`] = `
Object {
  "type.statusValue": "musit.observation.page.type.statusValueRequired",
}
`;

exports[`Validation.validateAlcohol should verify that volumeValue cannot be greater than 10 1`] = `
Object {
  "type.volumeValue": "musit.observation.page.type.volumeValueIncorrect",
}
`;

exports[`Validation.validateAlcohol should verify that volumeValue cannot contain dots 1`] = `
Object {
  "type.volumeValue": "musit.observation.page.type.volumeValueIncorrect",
}
`;

exports[`Validation.validateFromTo should not accept fromValue or toValue with dots as comma 1`] = `
Object {
  "type.fromValue": "musit.observation.page.type.fromValueIncorrect",
  "type.toValue": "musit.observation.page.type.toValueIncorrect",
}
`;

exports[`Validation.validateFromTo should validate correct values 1`] = `Object {}`;

exports[`Validation.validateFromTo should validate that commentsValue cannot be longer than 250 chars 1`] = `
Object {
  "type.commentValue": "musit.observation.page.type.commentsValueIncorrect",
}
`;

exports[`Validation.validateFromTo should verify that fromValue cannot be empty 1`] = `
Object {
  "type.fromValue": "musit.observation.page.type.fromValueRequired",
}
`;

exports[`Validation.validateFromTo should verify that fromValue cannot be empty space 1`] = `
Object {
  "type.fromValue": "musit.observation.page.type.fromValueIncorrect",
}
`;

exports[`Validation.validateFromTo should verify that fromValue cannot be null 1`] = `
Object {
  "type.fromValue": "musit.observation.page.type.fromValueRequired",
}
`;

exports[`Validation.validateFromTo should verify that toValue must be greater than fromValue 1`] = `
Object {
  "type.toValue": "musit.observation.page.type.toValueSmallerThanFromValue",
}
`;

exports[`Validation.validatePest should verify correct values 1`] = `Object {}`;

exports[`Validation.validatePest should verify that an observation with count and not lifecycle is not allowed 1`] = `
Object {
  "type.observations[0].observations.lifeCycleRequired": "musit.observation.page.type.lifeCycleRequired",
}
`;

exports[`Validation.validatePest should verify that an observation with count larger than 10 numbers is not allowed 1`] = `
Object {
  "type.observations[0].observations.count": "musit.observation.page.type.countIncorrect",
}
`;

exports[`Validation.validatePest should verify that an observation with empty lifeCycle is not allowed 1`] = `
Object {
  "type.observations[0].observations.lifeCycle": "musit.observation.page.type.observations.lifeCycleIncorrect",
}
`;

exports[`Validation.validatePest should verify that an observation with only lifecycle is allowed 1`] = `Object {}`;

exports[`Validation.validatePest should verify that commentValue cannot be longer than 250 chars 1`] = `
Object {
  "type.commentValue": "musit.observation.page.type.commentsValueIncorrect",
}
`;

exports[`Validation.validatePest should verify that identificationValue cannot be empty 1`] = `
Object {
  "type.identificationValue": "musit.observation.page.type.identificationValueRequired",
}
`;

exports[`Validation.validatePest should verify that identificationValue cannot be empty space 1`] = `
Object {
  "type.identificationValue": "musit.observation.page.type.identificationValueIncorrect",
}
`;

exports[`Validation.validatePest should verify that identificationValue cannot be null 1`] = `
Object {
  "type.identificationValue": "musit.observation.page.type.identificationValueRequired",
}
`;
